The Revolutionaries Rework

SlamBamActionman opened the pull request:
The one rework to rule them all.

Revolutionaries as a gamemode is currently fairly derided as a gamemode that relies too much on murderboning and being too "metagamey" to counterplay. This design document aims to remake the Revolutionary antag into one that escalates appropriately in presence throughout the round, has clear end conditions and doesn't encourage round removal. 

> "I want to see people kidnap the QM, lock him in a room with a bunch of communism posters and have 5 medical doctors wearing red headbands scream at him until he converts to the revolutionary cause"

Enjoy!

_I want to highlight the previously submitted Revolutionaries reworks #251 and #146, as some concepts from those documents have made it into this proposal where appropriate._

SirChopwood commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,259 @@
+Codewords are able to deal damage even when communicated across different mediums, however the conversion strength is reduced to compensate for how easy it may be to do so:
+
+- A Revolutionary speaking a codeword through a headset radio deals **0.25**x CHP damage.
+- Announcement messages with a codeword by a Revolutionary deals **0.5**x CHP damage, without being divided among the people hearing it.
```
Simple solution would be to add the location of the comms console to the announcements, which i know some forks on 13 already do anyway. "**Sent by: Urist McRev** at Science South Maints"

SlamBamActionman replied:
I think this is a fair point and even in other gamemodes it's considered a pain if the comms console goes missing and someone starts abusing it. The suggestion above to include the console's location would probably be good enough to counteract this.

goet commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,259 @@
+Deconversion puts you fully on the crew side. This does fully include the ability to sell out your fellow Revolutionaries by remembering their names. Headrevs therefore have a choice to act more stealthily disguise themselves and blend in with the Revolution, or assign another Revolutionary (possibly their first convert) to act as the revolution's leader to throw off potential deconvert sell-outs ...
+
+Methods of deconversion include:
+  - Harm baton, heals **10** CHP per hit
```
I feel like this makes a crew lose before the round ever reaches the open confrontation, arresting a single antag in a couple minutes is considered pretty fast normally, but would be deadly here. Might be fine with some number wrangling, but wanted to give an alternative:

I love the idea to nudge sec to be violent about this, possibly lethal after a certain point, but some nonviolent ways should be available to the rest of the crew. Perhaps being around the captain humanizes him so he passively heals CHP. Or alcohol/space drugs makes people content so it greatly increases the healing rate. The random pizza event deals global CHP damage, but actually eating it heals a ton. Stuff like this would actively push revs to be increasingly more public to counteract these effects. It could also make paranoid heads do the things corpo managers do to keep morale up, which could be pretty funny.

I agree that deconversion should not happen on accident and it makes these tricky, but they could work similarly to how poison works in a lot of games. Simply can't go over 0, and/or the effect scales in a way that can't deconvert, but the heal amount is greater the further away from 0 the person is.

Everturning commented:
this is a really good idea and it seems real fun. Here are the things I like and that I don't.

Like: 
- Mindshield being breakable. It makes it have some reason to buy multiple for the same people instead of just to distribute to crew. And, I love the scene you detailed in PR description.
- I just generally like it. there's a lot here I like. I cant organize it all. this is just a good rework proposal.

Dislike (mostly if not entirely to do with the DDD, which is a good idea in itself, just the items are weird)
- Cargommunism Pad is OP. It's just a evilcoded QM board, with supposedly little counterplay. Maybe it should be cargo locked or have at least one major difference from the digiboard.
- Mind-swap doesn't make sense. I understand it's locked behind tier 3, but... a complete mindswap? why would revolutionaries have access to wizard spells? it's weird.
- The defriebman should make it so that the people you defib are convertable, not instantly converted. Makes for a cool scene where when you wake up from death you have a few seconds to run before joining the wave.
- Something about the headset codewords thing irks me. I understand the intent, but mechanically it's very strong. Even 0.25 of the damage is still kind of powerful with the major codewords. But maybe it's not that. I still see *something* wrong there that I can't really define.

An idea I had: There should be some sort of interaction with the news management console. Maybe amp up the communications console effect on there? Would probably only apply if the crewmember reads the article.

Please take this all with a grain of salt, as, after all, these are just the musings of a player. I just wanted to add my thoughts on such a good rework proposal. This also completely remakes the conversion format for the game mode, which is truly... Revolutionary.

TL;DR, maybe analyze the DDD items more, mindshields being breakable is good, radio codewords needs a bit of looking at

BombasterDS commented:
This revolution idea sounds amazing. What are you thinking about renaming Conversion Health Points to Loyalty?
Also it should have interactions with reporter and news.

Kadeo64 commented:
wow holy shit this sounds 200% better

who would have thought that making murderbone a bad idea would solve revs

Kadeo64 commented:
`CentComm agents and other Admeme should be similarly unconvertable by default.`

*except centcomm interns :trollface:

BombasterDS commented:
> The Captain is also unique in that they always have 100 CHP and can never go below this.

Head of Security should have same property imo.

mirrorcult commented:
there are some very good concepts here

Djungelskog2 commented:
I like how not only did you stop the issues with the gamemode but actively turn some negatives such as metagaming pr lack of RP into positives by mechanically forcing people to socialise or do social deduction

CaelleakNavinski commented:
Yes. Ten thousand times yes. This is amazing. Very very well done.

SlamBamActionman commented:
> This revolution idea sounds amazing. What are you thinking about renaming Conversion Health Points to Loyalty? Also it should have interactions with reporter and news.

I did this, though I still call it Loyalty Health Points for the acronym LHP. Calling it just Loyalty works now though, and it sounds better (plus makes the Convertable state sound more distinct).

LittleNorthStar commented:
A potential idea I have is the implementation of propaganda videotapes which can be used on TV's to provide a area deconversion/conversion effect similar to posters but more powerful

also a idea for a T3 Dead Drop Items, IFF integrated surplus rifle and IFF rifle ammo schematic, essentially a special gun and ammo type which doesn't hit revolutionaries instead passing over them to hit other targets, individually the guns are kind of pathetic (low fire rate and damage) but in a group they can make for a good final push for the revolution without the normal risk of friendly fire you would get from just ordering guns from cargo

ArcaneOcto commented:
This seems incredible. Literally the perfect gamemode imo. There appears to be several strategies for victory on both sides which is awesome! Man, I really hope someone codes this. The gameplay sounds so much deeper than most of the gamemodes we have rn.

githubuser508 commented:
Small additional idea: If a Revolutionary kills themselves with damage (explosion, self-immolation), it could also perhaps deal LHP damage? I love the theming of this rework, and martyring oneself is another very "revolutionary" concept, I think.

saga3152 commented:
What if at tier 3 you could buy mosin and ammo blueprint? That's very commy. Maybe also some kind of revolution music you can play to increase loyalty damage and decrease loyalty overall

TytosB commented:
this is a bit of a personal thing, but i dont like that Revs seem to be going for an explicitly communist theme here. revolutions can be motivated by a number of factors, and it should be up to the players how they want to RP the flavor of their rebellion. one of the most fun times ive ever had as a headrev was being a chaplain that played it like a cult taking over the station by indoctrinating people.

ilovehans10 commented:
It would be interesting if sec/command killing a crew member also did damage to loyalty. It would mean that even if the game mode wasn't revs 2.0, but you instead became an initial rev via midround conversion, you would benefit from shitsec gunning down a protest earlier in the round.

southbridge-fur commented:
What is preventing this from being approved and merged? I'd like to start implementing this game mode but am hesitant due to this still being unmerged.

EthanQix commented:
A suggestion for that very last line : 
> Revs could be given objectives that, if successful, deals special map-wide LHP damage that is difficult to heal back up.

The trigger for this should be tied to converted Command members. Converted Command should also deal more Loyalty damage through the use of codewords than the common crew, maybe with a reduce malus when using radio/announcements.

SlamBamActionman commented:
I have updated the document with some small but impactful changes based on feedback:

1. Codewords over radio have been slightly lowered, and a clarification has been made that crew passively heal LHP. This means dealing damage over radio is slow if you're not using high-tier words, and if not constantly kept up can be negated by the passive heal. Crew also have the ability to negate this by simply removing headsets or selectively disabling certain telecom servers. Since the damage is also split among listeners it becomes better as a passive heal for Revolutionaries when the revolution is in full swing, rather than an attacking option.
2. Relaying a codeword over the announcement console only works once, but deals full damage. I don't love this option but it frees the console from being repeatedly abusable. 
3. Dealing/healing LHP damage upon death is now reserved for crew-on-crew, crew-on-rev, rev-on-crew and rev-on-rev violence. Mobs and other antags do not trigger the mechanic. This means Security don't have to worry about dealing lethally with other threats, but are punished for indiscriminately killing crew. Additionally, only a Revolutionary being killed by a crew is able to convert crew, to prevent Revolutionaries from spawning during a non-Rev round.*
4. Mindshields have been changed to not break, but instead loses its passive recharge when the owner is a Revolutionary. This means Security are not alerted if a converted department head no longer has a mindshield. However it might still be desirable to inject someone with a mindshield when deconverting them, as that provides a fresh 100 LHP shield, protecting them from new LHP damage.
5. Minor number adjustments (they're suggestions anyways) and text edits.

Everturning commented:
I must say I think there can be more to bring LHP.
An idea I had is having certain people become "loyalists" that are selected roundstart but don't actually activate until they are convinced (maybe also set a time gap of like 20 minutes to activate the loyalist's convertibility). They would gain a set of codewords to heal LHP, but considering how they would likely be near the revs when they're converted, they would have to make a run for it.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+Revolutionaries are a team conversion antag: a small number of players begin as antagonists, but are able to convert crew to antags, with the antag faction's strength largely being dependent on the number of conversion. As a conversion antag, a round with Revolutionaries should attempt to follow the stages below. 
+
+- Initial conversions
+  - This is the stage where the antag begin conversion crew - this should be done stealthily, and failing to do so may result in a swift win for crew. To their benefit the antag is given some kind of initial advantage, providing a quick power bump to get things going.
```
This is a constant problem with snowballing antags in general. The design of any Revs mode that involves snowballing is going to have to counterbalance that a fluked round (i.e. "Revs are discovered immediately through misfortune") with recovery.

The critical flaw of current Revs is the "discovery" mechanic. The only thing that prevents the HOS from creating a mindshield line in front of Sec every shift is admin rules; any Revs refactor must find a way of resolving this.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+### Win Conditions
+
+Win conditions for a round with a Revolutionary antag are the following:
```
So I guess the logic is that Revs is not about taking the station at all, but... just having RR'd or arrested/converted heads?

Why should Revs bother to convert heads? The difference is just major vs minor victory. 

If Heads know they can deny a Rev major victory, what stops Heads deliberately fleeing the station via a jetpack or a shuttle?

SlamBamActionman replied:
> Why should Revs bother to convert heads? 

Not only is a Command rev *very* useful if done quietly, they contribute to the DDD percentages as well. I disagree with the notion that players won't try for the major win as an initial position.

> If Heads know they can deny a Rev major victory, what stops Heads deliberately fleeing the station via a jetpack or a shuttle?

You'd need >50% of Command to do this to deny a Rev major. If Command insists on being shuttle cowards, Revs can still opt for a minor.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+- Revolutionary Minor Victory:  No *alive, unrestrained* Command *on the evac shuttle* are non-rev, <**50%** of Command are converted Revolutionaries.
+- Rev-Crew Tie: Both Rev Minor & Crew Minor criteria are met, or neither Rev Minor & Crew Minor criteria are met.
+- Crew Minor Victory: No *alive, unrestrained* headrevs are *on the evac shuttle* when Evac reaches CentComm.
+- Crew Major Victory: *All* revolutionaries are *deconverted* or *dead* when Evac reaches CentComm.
```
If you just need an alive HR and all Heads to be dead when you reach Centcomm, then this means the HRs can just bide their time and bomb the shit out of the evac/end-of-shift shuttle and completely ignore the conversion mechanic.

SlamBamActionman replied:
With what?

If you get to the point where a rev simply waits for evac, not only will they be fighting all of Sec, but they have to deal with actually killing all heads. Realistically this would need to be something unique like a chemist with CLF3 bottles, and there are pushes for making Evac bombing less lethal.

I don't entirely disagree with this point, but it is pretty much the same issue as the Traitors gamemode, and solutions can be applicable to both.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
````diff
@@ -0,0 +1,271 @@
+```
+Crew Major Victory! [Green text]
+All Revolutionaries were killed or deconverted! [Red text]
+```
````
Overall: the goals need reworking. A conversion antag is a "crew versus the bad guys" mode, just like nukies. Escaping on the shuttle at the end is the luxury of a loyalist crew (or a fleeing loyalist crew giving up the station to the mutineers). It makes more sense if the Revs are trying to claim the station, and thus they have to _remain_ on the station when the shuttle leaves, having killed the leadership they need to claim the station for... like, the Syndicate, but shhhh.

In general you can't just have the HRs have their greentext tied to the shuttle arrival at Centcomm moment, because:

1. It makes no sense. You overthrow the heads and then... go back to NT Centcomm?
2. It encourages gameplay centred on evac and the shuttle. The HRs need to kill or restrain the heads, the heads they need to kill or restrain are always going to be on the shuttle, so just bomb the shuttle. But because it's a Revs round rather than a Tot round, the station has no tots to actually cause chaos. That's a recipe for a long, boring round that ends with a shuttle bombing, which is everyones' least favourite SS14 rounds.

SlamBamActionman replied:
I just entirely disagree with this point because it is working under the previous Rev framing. Yes, the goals don't make sense if you perceive the intent of Revs to be "take over the station and free it from CC's clutches". But that is only if you choose that framing. 

"Start the revolution by conquering the station, and then head to CentCom to spread the revolutionary spirit to new stations! Don't let the NT loyalist scum get back home!" Put something like that in the antag description pop-up and you've made it clear that the goal revolves around eventually leaving the station.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+## Conversion
+
+Conversion is how the Revolutionary antag faction grows, and is a *requirement* to obtain Revolutionary Major Victory. To convert a crewmember they must be interacted with by a Revolutionary over time, and once a critera is met they can be converted to the Revolution. Crew are also able to de-convert Revolutionaries. Conversion is not restricted to Headrevs, though they have access to certain too ...
```
Designing around major victories is a trap. They're stretch goals, like blowing up the Nukie home.

SlamBamActionman replied:
Blowing up the Nukie home/CentCom isn't the standard major for Nukies. For crew, it is killing all Nuclear operatives. For Nukies, it is blowing up the station. I am of the opinion majors hold decent weight in that they become good goals to aim for, but not so far as to make a player endure boring rounds (e.g. if a single nukie is stalling, crew will try to hunt them down at first but then call evac and take the minor if it is becoming too boring).

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+Conversion is how the Revolutionary antag faction grows, and is a *requirement* to obtain Revolutionary Major Victory. To convert a crewmember they must be interacted with by a Revolutionary over time, and once a critera is met they can be converted to the Revolution. Crew are also able to de-convert Revolutionaries. Conversion is not restricted to Headrevs, though they have access to certain too ...
+
+To represent the convertable state a crewmember is in and the faction they belong to, there is now a hidden nummerical stat called Loyalty Health Points (LHP). 
```
As discussed, this is for all intents and purposes a mental health stat. The main struggle I ran into with my old equivalent was that it's a bit of a magic number for the crew, and hard to explain to people without making it really obvious it's a Revs round.

SlamBamActionman replied:
As of now, I am opposed to tying it into any mental health system, due to how integral it is to the gamemode.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+  - While a crewmember is in the Convertable state, they can be converted into a Revolutionary via specific actions available to Revolutionaries.
+  - When converted to a Revolutionary, their LHP is set to **-25**.
+  - A successful conversion is silent for surrounding players, however the player who is converted gets a pop-up and the appropriate music stinger.
+- Crew are able to "heal" LHP via specific actions, deconverting Revolutionaries into normal crewmembers.
```
It needs to be hard/impossible for a Rev to do this themselves, as otherwise carebear players will metagame themselves back to the crew's side.

Remember, Revs do not have to be players who have HR ticked, and could be a newbie or a hardcore bar RPer who hates all conversion modes. "Discovered" rounds often lead to people cryo-ing. The mode needs to work for players who don't want to be revs because it's not what they logged onto SS14 to do.

FairlySadPanda replied:
Appreciate this is super hard to get right, but "oh, I got made into a rev, time to go hug some secoffs and get whacked back into loyalty" or whatever is a problem to solve.

SlamBamActionman replied:
A rev player who doesn't want to play rev will always find a way to not play; whether that be through suiciding, cryoing or blatant deconversion. I am willing to try out harder coversions, but I'd rather mitigate it through other means (e.g. not being able to tell who is headrev).

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+The Captain is also unique in that they always have 100 LHP and can never go below this. This makes the Captain unconvertable to the Revolution, and they are the only Command on the station with this property. CentComm agents and other Admeme should be similarly unconvertable by default.
+
+Unlike normal damage, LHP can only be dealt/healed to an individual who is alive and not in crit. 
```
This is the usual conscious requirement we see in a lot of design, so it would presumably cover "players who are not dead, in crit or asleep".

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+Codewords are able to deal damage even when communicated across different mediums, however the conversion strength is reduced to compensate for how easy it may be to do so:
+
+- A Revolutionary speaking a codeword through a headset radio deals **0.2**x LHP damage. (Note that crew passively heal LHP, as described in the Deconversion section)
+- The first time a Revolutionary uses the Announcement console to send a message with a codeword, it deals **1**x LHP damage, without being divided among the people hearing it. This is a one-time trigger.
```
What should the Reporter role do?

FairlySadPanda replied:
Specifically spreading PDA messages laiden with disloyal subtext etc seems like a great thing for the reporter to do. Would be good to consider what roles you want to shine in this mode. E.g. the reporter, the bartender?

SlamBamActionman replied:
Social roles and departmental roles with a lot of communication have an inherent advantage. I wouldn't mind the news console being able to deal LHP damage, but on the other hand I wouldn't want to discourage players from reading the news any more than they do now...

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+Codewords are able to deal damage even when communicated across different mediums, however the conversion strength is reduced to compensate for how easy it may be to do so:
+
+- A Revolutionary speaking a codeword through a headset radio deals **0.2**x LHP damage. (Note that crew passively heal LHP, as described in the Deconversion section)
```
So the most obvious problem with this is that this falls over as soon as "deaf" is added as a trait. But otherwise it makes talking and hanging around near people a weapon for Revs. So if Revs is confirmed... you just take off your headset and stop allowing people to talk?

FairlySadPanda replied:
If we make it so that you can't talk in space, you could actually neuter Revs by just spacing the station.

SlamBamActionman replied:
For a confirmed revolution, popping out your headset key is an intended counter.

If you're deaf then, sure, conversion will be harder. If lip reading/sign language is a thing then this can be mitigated, and there are other methods of dealing LHP that don't involve speech.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+Being converted/deconverted by a character dying displays a special conversion message, to indicate it was the reason for being converted/deconverted.
+
+There is an interesting dichotomy that can form here where Revolutionaries have an easier time to "heal" revolutionary fervor by dealing LHP damage to each other via chanting codewords. On the other hand, Security has more access to stun weaponry and as such have an easier time subduing Revolutionary groups non-lethally. 
```
Or Sec can just kill everyone and let the Admins sort it out.

SlamBamActionman replied:
See: Death & LHP Damage

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+Authoritarian regimes killing of protesting civilians causing further unrest, or freedom fighters getting a shock when they first murder someone, are concepts that are closely tied to a revolution. They are also ideas that would be suitable mechanically to combat murder sprees from either side during a Revolution round. The following section is only relevant for crew and Revs; non-crew characters ...
+
+If a Revolutionary *or crewmember* dies by the hand of another crewmember (being dealt the damage that put them in a critical state), they will deal **15** non-split LHP in a **8** tile line-of-sight radius around their body. While remaining dead they will also provide a passive **-0.1** LHP / second in a **4** tile line-of-sight radius. This works in the same way but in reverse if a Revolutionar ...
```
This is sort of a complex idea so I'll try and make sure I have it straight: crewmembers take LHP when they see shocking things happen, but Revs will gain LHP when this happens. This can cause flips, and if it's Revs->Crew then it's a 1s stun?

SlamBamActionman replied:
Kill the other side: bad for your LHP, good for the other side's LHP
Kill your own side: bad for your LHP, good for the other side's LHP

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+There is an interesting dichotomy that can form here where Revolutionaries have an easier time to "heal" revolutionary fervor by dealing LHP damage to each other via chanting codewords. On the other hand, Security has more access to stun weaponry and as such have an easier time subduing Revolutionary groups non-lethally. 
+
+### Mindshields
```
This design means you can create antag Sec and Head roles (other than the Captain), a known hard design to get right in SS13/14.

A HR that can capture a HOS or Warden for a few minutes and convert them is going to power spike massively - that answers "why bother converting", I suppose - but in general Rev heads/secoffs feels like a problem.

SlamBamActionman replied:
With the de-emphasis on killing, a rev SecOff is gonna be very useful but not *as* impactful as they would be in current revs. Being able to smuggle out stuns and stuff is great, of course, but that should be a worthy reward for converting a protected and armed crewmember.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+- A newly injected mindshield always has full LHP, regardless of Revolutionary status.
+- Mindshields can be removed, much like any other implant. The exceptions for this is the Captain and ERT/CentComm, who have unremovable mindshields (however due to being unconvertable, this is mostly just flavor). 
+
+New mindshields are able to be purchased from Cargo at a reasonable price. One possible suggestion would be 4 implanters for $8000.
```
I think giving perfect mindshielding to the Captain, HOS, Warden, Dets, Secoffs and Cadets and then just not mindshielding the other heads (or giving them these weaker ones) makes more sense to avoid shithousery. Arguably HOP too given how powerful AA is.

FairlySadPanda replied:
Overall I'm really not a fan of keeping mindshields around at all anyway. It's an unsatisfying and slow mechanic and still requires "rev discovery" admin metashield rules that the admins hate enforcing. Mindshielding as an implant is just kinda bad design.

Feels more like something that could be done via surgery tho.

SlamBamActionman replied:
I disagree with this paragraph. Yes; Sec still benefit from mindshielding crew, but it is no longer a binary guarantee. Additionally, a Rev could easily get a mindshield themselves and become a trusted ally, or the revs successfully convett someone with a shield - which would also be why Sec would no longer be able to rely on the hud icon exclusively.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+- Posters can be taken down, and if recycled gives back some cloth.
+- Destroying a poster that is rolled up on a wall heals **5** LHP if it's a Revolutionary poster, or deals **10** LHP if it's a Loyal poster.
+
+### Dead Drop Duffelbag / The Revolutionaries Uplink 
```
Skimming over this section but there's some great ideas in here. Especially love the megaphone.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+- Headband is a headwear item that can be created by using a sharp object on a bandana. (Crafting method may be changed if dyeing clothes is implemented).
+- This is a fashion accessory accessible to all crew, however when worn by a Revolutionary their codewords deal 1.5x more LHP damage.
+
+**Revolutionary posters**
```
I'm biased as this mirrors my proposal. Think codewords and propaganda would make an equal pair of ways to wage LHP war so I'd bump this up in terms of importance.

FairlySadPanda reviewed (commented):
I'll probably sound quite harsh in some of these comments but it's mostly just trying to attack the idea so it can be defended and built out on. I do think the only major worries I'd have would be:

1. Making the mode too codeword heavy. I think you've got some pillars here - causing chaos and violence to weaken the crew's loyalty, spreading verbal propaganda, and spreading visual propaganda. Those I think are three pillars, going for any one of the three alone as the "main" way to convert people IMO would be problematic.
2. The escape shuttle gameplay needs revising away from. It's a dead end; you want to encourage a war on the station, not a dull round ending with a shuttlebombing for HRs to greentext on.
3. Lack of consideration of how the different departments interact with the mode. You need to at least be able to describe "why this mode is fun" for the departments, what each department can bring to the game mode, etc. 
4. Lack of consideration about the admin aspect of the mode and how to get rid of the admin metashielding rules.

SlamBamActionman commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+Revolutionary antags that start out this way are considered Head Revolutionaries (HeadRevs). A Head Revolutionary has the duty to begin the conversion of other crew and have some extra tools at their disposal to strengthen the progression of the antag faction, but they are not strictly necessary for the revolution antags to succeed. The goal for Head Revolutionaries is to convert crew to grow the ...
+
+Head Revolutionaries share the same team and when they are given the role they obtain the names and jobs of the other Head Revolutionaries on the station. Revolutionaries can be identified by other revolutionaries via an icon next to their character, however Head Revs do not have a unique icon.
```
I am not opposed to making the R blue for headrevs to see each other, as headrevs get each others' names roundstart.

With how the conversion mechanics have been changed and the core problem of "one Cargo Tech gets deconverted, rats out all the revolutionary leaders" is something I want to avoid.

Head Revs have ways to reveal themselves to their revolutionaries anyhow, by either just saying it or bringing out the DDD. This puts the choice into their hands.

SlamBamActionman commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+
+Head Revolutionaries share the same team and when they are given the role they obtain the names and jobs of the other Head Revolutionaries on the station. Revolutionaries can be identified by other revolutionaries via an icon next to their character, however Head Revs do not have a unique icon.
+
+For crew, the goal is to deconvert revolutionary converts, protect Heads from being converted, and ensure Head Revs are discovered and/or do not make it onto the evacuation shuttle. 
```
> the Revs shouldn't ever want to evac.

I feel this line comes from the poor framing of current Revs that they are supposed to "take over the station, period". If we take that to its full extent, it means you have a round where a major antag won and now... You just sit around? Unlike Traitors, which are destructive by nature, there is nothing much intrinsic to Revs (old or new) that encourages making the environment uninhabitable for Evac.

Making Revs have a win condition revolving Evac means you break the stalemate-ness, too. Command hiding in space? Who cares, just evac and let them rot in space while you party with a major/minor.

SlamBamActionman commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+## New Revolutionaries, Basic layout
+<sup>Any values given here are suggestions and are especially marked by being **bolded**. Any such value should be adjusted based on playtesting.</sup>
+
+The Revolutionary antag begins either as a 2-4 roundstart antag roll for the Revolutionary gamemode, or a single crewmember secretly becomes a Revolutionary antag during another gamemode midround.
```
Either way goes to be honest. Unlike Thieves and current Revs, you are not given any roundstart equipment. That is the main reason those roles are provided roundstart.

Pumkin69 commented:
DDD is a fun and interesting way of doing antag items
A couple things to note
-Mindshields seem a little underpowered
-Headbands should also be replaced in cargos bandana request, as i feel like it brings more light to them and also makes sense for cargo to be doing this, as a couple of other requests are usually filled through crafting, such as the knives one usually being filled by shivs and the baseball bats also being crafted.

Overall i really like the direction this is setting the revolutionaries gamemode as it gets it away from full murderbone revbay town and into more RP based and interesting fun town. Great concepts over all i love a lot of the items especially the megaphone, Headset override, Revolutionary stamp and the Mindswap. Good work i think this should be merged.

ilovehans10 commented on src/en/space-station-14/round-flow/proposals/revolutionaries-codeword-rework.md:
```diff
@@ -0,0 +1,271 @@
+  - The antag faction has been exposed and crew are now working to counteract the antags. Depending on how well the two previous stages have gone the crew may be in either a good or poor position to fight back.
+- Climax
+  - An encounter between the antags and crew that largely decides the outcome of the round. 
+- Resolution
+  - After the climax the round finishes up, resolving the round for remaining players. Usually this is the last few minutes before the evacuation shuttle arrives.
```
It could be interesting if there was a sort of capture the flag aspect to the end of the round. The heads think they have weeded out most of the Revs so they go to the bridge and hit a facility secured button telling centcom that they think the station is secured. This starts a ten minute timer and during this time either command or Revs can capture key areas on the station via some means. At the end of the station secured timer the round ends and it's revealed who has control of the station.

There are some real kinks with this idea like what incentives heads to hit the button, where are the key points on the station that need securing, and what if all heads are killed before that point. This does have the upside of no shuttle and no need for TDM.

SlamBamActionman commented:
After maintainer discussion the decision has been made to merge this design document. 

While there are concerns about the particulars of the gamemode and how it relates to other current and future conversion antags (most notably Cults), there is interest to get this rework merged for two reasons:

a) Current Revs is in an undesirable state and is seen as fairly broken from both a gameplay and admin perspective. There are proposed bandaid fixes, and by having this design document the maintainer team can get those fixes merged in the interim without worrying about a rework being stalled by discussion. 

b) Getting a rework of this magnitude into the hands of players to playtest is important to ensure the final version ends up in a good place, and this will allow work on a playtest version to begin. 

The next step to begin implementing this gamemode is likely to depend on the Chat Refactor I am currently heading, as it relies on being able to process words sent via messages (while current code supports this to some degrees with things like codewords, that is implemented in a largely undesirable manner codewise). Some areas that can be independently worked on however:
- Loyalty hitpoints and conversion, dealing and receiving LHP damage from non-word sources 
- Compiling a list of conversion words appropriate for the different tiers
- Headbands & posters
- The DDD interface and interactions
- Win conditions

Maintainers Bhjin & Myr and Julian have given concurring approval on Discord to merge this document.

illdoodle commented:
Thanks for mentioning my rework concept!
The LHP idea is really cool, but I think the deconversion/conversion conditions are a bit too harsh (especially the moment with the harm baton). This can lead to crew members constantly changing sides, as there are many ways to change LHPs. However, it's a really good concept and I hope someone brings it to life someday. So far the only problem is numbers and balance, which I think can be polished when implemented into the game.

walksanatora commented:
the main blocker right now for implemntation IIRC is the chatcode refactor. with this being a codeword rework it requires decent chatcode integration.

northerndakotas commented:
Two men walk into a bar.
The bartender says "Hi"
Both men need a minute to recover from the damage this has caused.

Okay jokes aside, I'll leave my thoughts here in preparation for the eventual chat refactor. Starting with the issue that prompted the above joke.

1. LHP damage values. I'm aware that the final values of these codewords are likely going to change. That said, my assumption is that codewords are supposed to do damage roughly proportional to how suspicious they sound in conversation. With that in mind, how many times more suspicious is the word communism in regular conversation than the word hello? Well it's infinitely more suspicious. But with the current damage values saying hello 3 times does just as much damage as comrade. If a codeword is going to be THAT imperceptible, it needs to do basically no meaningful damage. I also don't think dealing LHP damage should pause passive loyalty healing. If a rev can't out damage 0.1/second then they probably don't deserve to convert that person. At the very least there should be a damage threshold. For example, one rev saying tier one words over common radio dealing 0.01 LHP damage every ten seconds should not be able to shut down the passive healing for the whole station forever. That however brings me to a question about the codeword cooldown.

2. Is the 10 second cooldown on the speaker or the hearer? I assume it's the speaker who goes on cooldown when they use a codeword as that, I think, would cause less problems gameplay wise. However, if I am misunderstanding and it's a cooldown that starts when you hear a codeword and prevents you from taking more damage for 10 seconds, that would cause some very big problems. I won't go into that more right now as I'm pretty sure it's the other way, but I can give an example if it is a cooldown on the hearer.

3. Deconversion stun. This feels like a weird thing to retain as part of the game mode. Since conversion is done with words instead of a men-in-black style mind control device, and deconversion is down with good old fashioned violent oppression instead of a magic mind control blocking device, I thought we were moving away from the literal mind control interpretation of how Revs work. This seems to only exist as a way to give the crew undeniable proof that there are revs, but the deconverted rev could just talk to them, and more interestingly, they could lie about being deconverted. To me it's unnecessary and gamifies something that could be handled with RP.

4. Discoverability. This is actually my biggest question in regards to the whole game mode: How does the crew know there is a revolution? The tiers of codewords are described to have varying levels of discoverability, but why does that even matter? Say Sec hears you saying "revolutionary words", at what point can they start arresting people? This is especially important since deconversion requires not only arresting people, but also keeping them in cuffs for long periods of time and intentionally harmbattoning them, both of which are against space law and would probably get you a roleban if done now. So Sec has to break the rules to combat revs for the crime of... saying words? How would this be added to space law: Tier 5 crime, Mutiny: Saying words that sound vaguely communist. 
Musician: "Anyway, here's Viva La Vida." Does Sec get to beat the musician to death for playing his "new favorite" song or does plausible deniability protect a rev indefinitely? Similarly, if a non-rev clown decides to publicly say they love communism, does sec then go full 1984 because they heard a codeword? I think it would help if the crew had less effective, but also less violent ways to restore crew loyalty. Which brings me to...

5. Heads of Staff. As it stands right now, command has basically no impact on the round other than starting with a bit more LHP and being a target. I think it would be better both thematically and for gameplay if heads had a more impactful role. First of all, I would make it so that heads of staff would give passive LHP healing to their own department staff when in proximity to them, including to revs that work for them. Something like 0.5 LHP/second. On a flipside, a converted head of staff would instead drain the LHP of their staff, making heads a very worthwhile target for revs to convert or kill despite how difficult it might be. The idea is that spending time working alongside your boss would make you less likely to want to revolt against them. It also means that brand new revs will be discouraged from going right back to work and instead will want to gather with other revs and build up each other's disloyalty, only venturing back into their own department when they're sure they are zealous enough to withstand the pressure of being around their supervisor or when there are enough of them to cover each other. This, I think, encourages "rev like" behavior and makes them more discoverable than just some people saying some words. There also is nice RP potential of a captured rev in jail having their boss come talk to them through the bars pleading with them to give up this revolutionary nonsense. This change would also make some departments natural stongholds for the crew, and others natural weaknesses. For instance medical, sec, and some parts of sci would be hard to take over, while service, cargo, and engi would most likely be the first ones to fall. Which honestly feels like a very white collar/blue collar split that makes a lot of sense thematically. Finally, I think this change would have a positive impact on non-rev rounds as it would encourage all heads of staff to be actively involved with their department, and not just the guy with the best tools that goes off and does his own thing for the whole round.
